### DISCUSSION TITLE
Migration guide from v2 to v3

### DISCUSSION THREAD

**Discussion by hossammourad** [asciinema/discussions#228] [Q&A]
I'm working on a codebase that's on v2 of the package, and I want to upgrade to v3, is there any breaking changes between the two versions I should be aware of? any guide that I can use to facilitate the migration? I looked up the README and the existing issues but found nothing.

I'm currently using v2 as following:

```js
asciinema.player.js.CreatePlayer(this.ref, this.props.src, this.props);
asciinema.player.js.UnmountPlayer(this.ref);
```

What's the equivalent of that in v3?

**Comment by ku1ik** [comment]
Right, v2-v3 migration doc would be nice indeed. Meanwhile translation would be as follows:

```javascript
import * as AsciinemaPlayer from 'asciinema-player';
const player = AsciinemaPlayer.create(this.props.src, this.ref, this.props);
player.dispose();
```

**Comment by ku1ik** [comment]
I created the upgrade guide here: https://docs.asciinema.org/manual/player/upgrading/#upgrading-from-v2-to-v3
